### PR TITLE
Default naming feature for azapi resource

### PR DESCRIPTION
**Pull request by xiaxyi**
acc test:
```
--- PASS: TestAccGenericResource_defaultsNamingPrefixAndSuffix (365.82s)
--- PASS: TestAccGenericResource_defaultsNaming (580.12s)
PASS
ok      github.com/Azure/terraform-provider-azapi/internal/services     580.778s

--- PASS: TestAccGenericResource_defaultsNamingRemovingChars (379.38s)
PASS
ok      github.com/Azure/terraform-provider-azapi/internal/services     380.092s


```

### CODE REVIEW

File: internal/provider/provider.go [hunk 1]
```diff
@@ -156,6 +157,14 @@ func azureProvider() *schema.Provider {
 				Description: "This will disable the Terraform Partner ID which is used if a custom `partner_id` isn't specified.",
 			},
 
+			"default_name": resourceName.SchemaResourceName(),
+
+			"default_naming_prefix": resourceName.SchemaResourceNamePrefix(),
+
+			"default_naming_suffix": resourceName.SchemaResourceNameSuffix(),
+
+			"naming_caf_enabled": resourceName.SchemaResourceNameCafEnabled(),
```
Comment (ms-henglu) [inline internal/provider/provider.go:166]: Can we remove this field and its schema, since it's not used.

**Review by ms-henglu — CHANGES_REQUESTED** [review]
hi @xiaxyi 

Thanks for this PR - I've taken a look through and left some comments. And please also add document and test cases for this new feature.

If we can make those changes then we should be able to get this merged.

Thanks!

File: internal/provider/provider.go (hunk 1 above)
Comment (xiaxyi) [reply internal/provider/provider.go]: removed.

**Comment by xiaxyi** [comment]
Thanks @ms-henglu for reviewing this pr. Updated the code.

**Review by ms-henglu — APPROVED** [review]
LGTM!

**Pull request merged**